lib/FrontendTool/FrontendTool.cpp

```diff
@@ -184,20 +185,39 @@ static bool emitMakeDependenciesIfNeeded(DiagnosticEngine &diags,
 }
 
 namespace {
+struct SwiftModuleTraceInfo {
+  std::string ModulePath;
+  bool IsImportedDirectly;
+  bool IsCompiledWithLibraryEvolution;
+};
+
 struct LoadedModuleTraceFormat {
   std::string Name;
   std::string Arch;
-  std::vector<std::string> SwiftModules;
+  std::vector<SwiftModuleTraceInfo> SwiftModules;
 };
 }
 
 namespace swift {
 namespace json {
+template <> struct ObjectTraits<SwiftModuleTraceInfo> {
+  static void mapping(Output &out, SwiftModuleTraceInfo &contents) {
+    out.mapRequired("path", contents.ModulePath);
+    out.mapRequired("is-imported-directly", contents.IsImportedDirectly);
+    out.mapRequired("is-compiled-with-library-evolution",
```

Comment:
Let's call this one "library evolution" or "supports library evolution". (I feel like your initial name makes it sound like it's incidental information about how it was compiled, rather than something that can affect clients.)

Reply:
I tried to make a naming scheme such that we can guess the types based on the keys themselves. `supportsLibraryEvolution` sounds like it could be a string if you don't know the details (e.g. yes/no/partially). Do you think having some consistency for keys<->types is worthwhile or not that useful?

Reply:
I think that's a reasonable principle, but I don't think a strict `is` really provides that much more certainly over a predicate-conjugated `supports`. Let's go with `supportsLibraryEvolution` then.